tools/c7n_mailer/c7n_mailer/sendgrid_delivery.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import sendgrid
+import six
+from sendgrid.helpers.mail import Email, Content, Mail
+
+from .utils import (
+    get_message_subject, get_rendered_jinja)
+
+
+class SendGridDelivery(object):
+
+    def __init__(self, config, logger):
+        self.config = config
+        self.logger = logger
+        self.sendgrid_client = sendgrid.SendGridAPIClient(apikey=os.environ.get('SENDGRID_API_KEY'))
+
+    def get_to_addrs_sendgrid_messages_map(self, queue_message):
```

Comment:
this function and the get_message_content function should be moved into a base class that SendGridDelivery/SqsDelivery/etc should all inherit from, these two funcs get repeated quite a bit in all parts of the mailer code. Not saying it should happen in this PR, but something to consider